pkg/hostap/monitor.go

```diff
@@ -98,7 +99,18 @@ func (m *InterfaceMonitor) StartMonitor() error {
 	m.stopWg.Add(4)
 	m.deauthRequests = make(map[string]int64)
 	m.ifEventCh = make(chan netlink.LinkUpdate)
-	m.IfEventHandler.Subscribe(m.IfName, m.ifEventCh)
+	pfInfo, err := trafficcontrol.GetSriovPFInfo(m.IfName)
+	if err != nil {
+		return err
+	}
+	for _, vfInfo := range pfInfo.VFs {
+		err = trafficcontrol.ConfigureVF(m.Logger, pfInfo, vfInfo)
```

Comment:
I see this model around a lot - monitor.go looping over the VFs.

Would it be cleaner to just pass around the PFInfo struct and have the trafficcontrol code do the looping?

Do we want to make the PFInfo more of an object-oriented object that owns its own actions?

Reply:
yes, correct. now tied appropriate methods to PFInfo/VFInfo structs and using it.